bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/api/AgsMagnitude.scala

```diff
@@ -31,10 +28,8 @@ object AgsMagnitude {
    * Gets the widest possible range limits incorporating the given conditions
    * and speeds.
    */
-  def rangeLimits(mc: MagnitudeCalc, c1: (Conditions, GuideSpeed), c2: (Conditions, GuideSpeed)): MagnitudeConstraints = {
-    val ml = mc(c1._1, c1._2).union(mc(c2._1, c2._2)).getValue
-    MagnitudeConstraints(ml.getBand.toNewModel, FaintnessConstraint(ml.getFaintnessLimit.getBrightness), ml.getSaturationLimit.asScalaOpt.map(s => SaturationConstraint(s.getBrightness)))
-  }
+  def rangeLimits(mc: MagnitudeCalc, c1: (Conditions, GuideSpeed), c2: (Conditions, GuideSpeed)): Option[MagnitudeConstraints] =
+    mc(c1._1, c1._2).union(mc(c2._1, c2._2))
```

Comment:
A bit of thinking out loud:  in general, `MagnitudeConstraint.union`  has got to produce an `Option` since you don't know whether they contain the same bands.  Here it *seems* like a safe assumption that a given `MagnitudeCalc` isn't going to produce `MagnitudeConstraints` with different bands given different `Conditions` and `GuideSpeed`.  In fact we make that assumption in the `band` method below.

In general, I think the `union` method could be made to work even if there is no saturation limit or if one of the two constraints being compared doesn't define saturation limits.  In those cases, the union could just be the one with the faintest faintness limit.  If both define a saturation limit, then it takes the max as it does now.  Basically think of no saturation limit as a limit of negative infinity.

So does this mean we should make this method private, call `union` and assume the value is defined?  That doesn't make me feel good either. :-/  I think this is a sign that the API is not perfect.